### PR TITLE
HBASE-24974: Provide a flexibility to print only row key and filter for multiple tables in the WALPrettyPrinter

### PR DESCRIPTION
**Pull request by sandeepvinayak**
backport of https://github.com/apache/hbase/pull/2345
@bharathv @virajjasani

### CODE REVIEW

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   1m 10s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| -0 :warning: |  test4tests  |   0m  0s |  The patch doesn't appear to include any new or modified tests. Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch.  |
||| _ branch-1 Compile Tests _ |
| +1 :green_heart: |  mvninstall  |  10m 12s |  branch-1 passed  |
| +1 :green_heart: |  compile  |   0m 43s |  branch-1 passed with JDK v1.8.0_262  |
| +1 :green_heart: |  compile  |   0m 49s |  branch-1 passed with JDK v1.7.0_272  |
| +1 :green_heart: |  checkstyle  |   1m 59s |  branch-1 passed  |
| +1 :green_heart: |  shadedjars  |   3m 18s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   0m 48s |  branch-1 passed with JDK v1.8.0_262  |
| +1 :green_heart: |  javadoc  |   0m 43s |  branch-1 passed with JDK v1.7.0_272  |
| +0 :ok: |  spotbugs  |   3m 16s |  Used deprecated FindBugs config; considering switching to SpotBugs.  |
| +1 :green_heart: |  findbugs  |   3m 13s |  branch-1 passed  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   2m  5s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 43s |  the patch passed with JDK v1.8.0_262  |
| +1 :green_heart: |  javac  |   0m 43s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 48s |  the patch passed with JDK v1.7.0_272  |
| +1 :green_heart: |  javac  |   0m 48s |  the patch passed  |
| -1 :x: |  checkstyle  |   1m 44s |  hbase-server: The patch generated 1 new + 22 unchanged - 2 fixed = 23 total (was 24)  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  shadedjars  |   3m  4s |  patch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  hadoopcheck  |   5m  2s |  Patch does not cause any errors with Hadoop 2.8.5 2.9.2.  |
| +1 :green_heart: |  javadoc  |   0m 32s |  the patch passed with JDK v1.8.0_262  |
| +1 :green_heart: |  javadoc  |   0m 43s |  the patch passed with JDK v1.7.0_272  |
| +1 :green_heart: |  findbugs  |   3m  9s |  the patch passed  |
||| _ Other Tests _ |
| -1 :x: |  unit  | 160m 57s |  hbase-server in the patch failed.  |
| +1 :green_heart: |  asflicense  |   0m 31s |  The patch does not generate ASF License warnings.  |
|  |   | 206m  2s |   |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hadoop.hbase.mapreduce.TestLoadIncrementalHFilesUseSecurityEndPoint |
|   | hadoop.hbase.mapreduce.TestLoadIncrementalHFiles |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.12 Server=19.03.12 base: https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2389/1/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/2389 |
| JIRA Issue | HBASE-24974 |
| Optional Tests | dupname asflicense javac javadoc unit spotbugs findbugs shadedjars hadoopcheck hbaseanti checkstyle compile |
| uname | Linux 61fd683066f0 4.15.0-112-generic #113-Ubuntu SMP Thu Jul 9 23:41:39 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | /home/jenkins/jenkins-home/workspace/Base-PreCommit-GitHub-PR_PR-2389/out/precommit/personality/provided.sh |
| git revision | branch-1 / 624fd04 |
| Default Java | 1.7.0_272 |
| Multi-JDK versions | /usr/lib/jvm/zulu-8-amd64:1.8.0_262 /usr/lib/jvm/zulu-7-amd64:1.7.0_272 |
| checkstyle | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2389/1/artifact/out/diff-checkstyle-hbase-server.txt |
| unit | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2389/1/artifact/out/patch-unit-hbase-server.txt |
|  Test Results | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2389/1/testReport/ |
| Max. process+thread count | 4298 (vs. ulimit of 10000) |
| modules | C: hbase-server U: hbase-server |
| Console output | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2389/1/console |
| versions | git=1.9.1 maven=3.0.5 findbugs=3.0.1 |
| Powered by | Apache Yetus 0.11.1 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   1m  6s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| -0 :warning: |  test4tests  |   0m  0s |  The patch doesn't appear to include any new or modified tests. Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch.  |
||| _ branch-1 Compile Tests _ |
| +1 :green_heart: |  mvninstall  |  10m  6s |  branch-1 passed  |
| +1 :green_heart: |  compile  |   0m 42s |  branch-1 passed with JDK v1.8.0_262  |
| +1 :green_heart: |  compile  |   0m 50s |  branch-1 passed with JDK v1.7.0_272  |
| +1 :green_heart: |  checkstyle  |   1m 56s |  branch-1 passed  |
| +1 :green_heart: |  shadedjars  |   3m 16s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   0m 48s |  branch-1 passed with JDK v1.8.0_262  |
| +1 :green_heart: |  javadoc  |   0m 43s |  branch-1 passed with JDK v1.7.0_272  |
| +0 :ok: |  spotbugs  |   3m 17s |  Used deprecated FindBugs config; considering switching to SpotBugs.  |
| +1 :green_heart: |  findbugs  |   3m 13s |  branch-1 passed  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   2m  3s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 41s |  the patch passed with JDK v1.8.0_262  |
| +1 :green_heart: |  javac  |   0m 41s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 49s |  the patch passed with JDK v1.7.0_272  |
| +1 :green_heart: |  javac  |   0m 49s |  the patch passed  |
| +1 :green_heart: |  checkstyle  |   1m 46s |  hbase-server: The patch generated 0 new + 22 unchanged - 2 fixed = 22 total (was 24)  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  shadedjars  |   3m  4s |  patch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  hadoopcheck  |   5m  2s |  Patch does not cause any errors with Hadoop 2.8.5 2.9.2.  |
| +1 :green_heart: |  javadoc  |   0m 38s |  the patch passed with JDK v1.8.0_262  |
| +1 :green_heart: |  javadoc  |   0m 46s |  the patch passed with JDK v1.7.0_272  |
| +1 :green_heart: |  findbugs  |   3m 10s |  the patch passed  |
||| _ Other Tests _ |
| -1 :x: |  unit  | 161m 53s |  hbase-server in the patch failed.  |
| +1 :green_heart: |  asflicense  |   0m 31s |  The patch does not generate ASF License warnings.  |
|  |   | 207m 10s |   |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hadoop.hbase.mapreduce.TestLoadIncrementalHFilesUseSecurityEndPoint |
|   | hadoop.hbase.mapred.TestTableSnapshotInputFormat |
|   | hadoop.hbase.mapreduce.TestSecureLoadIncrementalHFiles |
|   | hadoop.hbase.client.TestAdmin1 |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.12 Server=19.03.12 base: https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2389/2/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/2389 |
| JIRA Issue | HBASE-24974 |
| Optional Tests | dupname asflicense javac javadoc unit spotbugs findbugs shadedjars hadoopcheck hbaseanti checkstyle compile |
| uname | Linux 109d42f4f47f 4.15.0-112-generic #113-Ubuntu SMP Thu Jul 9 23:41:39 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | /home/jenkins/jenkins-home/workspace/Base-PreCommit-GitHub-PR_PR-2389/out/precommit/personality/provided.sh |
| git revision | branch-1 / 624fd04 |
| Default Java | 1.7.0_272 |
| Multi-JDK versions | /usr/lib/jvm/zulu-8-amd64:1.8.0_262 /usr/lib/jvm/zulu-7-amd64:1.7.0_272 |
| unit | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2389/2/artifact/out/patch-unit-hbase-server.txt |
|  Test Results | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2389/2/testReport/ |
| Max. process+thread count | 4348 (vs. ulimit of 10000) |
| modules | C: hbase-server U: hbase-server |
| Console output | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2389/2/console |
| versions | git=1.9.1 maven=3.0.5 findbugs=3.0.1 |
| Powered by | Apache Yetus 0.11.1 https://yetus.apache.org |


This message was automatically generated.

File: hbase-server/src/main/java/org/apache/hadoop/hbase/wal/WALPrettyPrinter.java [hunk 1]
```diff
@@ -223,6 +218,22 @@ public void setRowFilter(String row) {
...
+  public void setOutputOnlyRowKey() {
+    this.outputOnlyRowKey = true;
+  }
+
+  /**
+   * sets the position to start seeking the WAL file
+   * @param position
+   *          initial position to start seeking the given WAL file
+   */
+  public void setPosition(long position) {
```
Comment (bharathv) [inline hbase-server/src/main/java/org/apache/hadoop/hbase/wal/WALPrettyPrinter.java:233]: This code seems to be from HBASE-21185. Mind cherry-picking that in a separate commit (in the same PR). Easier to track it that way.

**Review by bharathv — CHANGES_REQUESTED** [review]

File: hbase-server/src/main/java/org/apache/hadoop/hbase/wal/WALPrettyPrinter.java (hunk 1 above)
Comment (sandeepvinayak) [reply hbase-server/src/main/java/org/apache/hadoop/hbase/wal/WALPrettyPrinter.java:233]: @bharathv I didn't add this method or position as part of this commit, it is already there at the end, I just added a java doc and repositioned it just like the master. Do you still want to have it as a separate commit?

Comment (bharathv) [reply hbase-server/src/main/java/org/apache/hadoop/hbase/wal/WALPrettyPrinter.java:233]: Ah sorry, I missed that part. No worries then.

**Review by bharathv — APPROVED** [review]

**Pull request merged**